core/domain/user_query_domain.py

```diff
@@ -0,0 +1,161 @@
+            ValidationError. Expected user ID in user_ids to be a valid user ID.
+            ValidationError. Expected sent_email_model_id to be a string.
+        """
+        if not isinstance(self.id, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected ID to be a string, received %s' % self.id)
+
+        if not isinstance(self.params, tuple):
+            raise utils.ValidationError(
+                'Expected params to be of type tuple, received %s'
+                % type(self.params))
+
+        if not isinstance(self.submitter_id, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected submitter ID to be a string, received %s' %
+                self.submitter_id)
+        if not utils.is_user_id_valid(self.submitter_id):
+            raise utils.ValidationError(
+                'Expected submitter ID to be a valid user ID, received %s' %
+                self.submitter_id)
+
+        if not isinstance(self.status, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected status to be a string, received %s' % self.status)
+        if self.status not in feconf.ALLOWED_USER_QUERY_STATUSES:
+            raise utils.ValidationError('Invalid status: %s' % self.status)
+
+        if not isinstance(self.user_ids, list):
+            raise utils.ValidationError(
+                'Expected user_ids to be a list, received %s' %
+                type(self.user_ids))
+        for user_id in self.user_ids:
+            if not isinstance(user_id, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Expected each user ID in user_ids to be a string, '
+                    'received %s' % user_id)
+
+            if not utils.is_user_id_valid(user_id):
+                raise utils.ValidationError(
+                    'Expected user ID in user_ids to be a valid user ID, '
+                    'received %s' % user_id)
+
+        if self.sent_email_model_id and not isinstance(
+                self.sent_email_model_id, python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected sent_email_model_id to be a string, received %s'
+                % self.sent_email_model_id)
+
+    @classmethod
+    def create_default(cls, query_id, query_params, submitter_id):
+        """Create default user query.
+
+        Args:
+            query_id: str. The id of the query.
+            query_params: UserQueryParams. The params of this query.
+            submitter_id: str. The ID of the user that submitted the query.
+
+        Returns:
+            UserQuery. The default user query.
+        """
+        return cls(
+            query_id, query_params, submitter_id,
+            feconf.USER_QUERY_STATUS_PROCESSING, []
+        )
+
+    def delete(self, sent_email_model_id=None):
+        """Archive the query.
```

Comment:
I'm confused. The method name is delete, the docstring says archive, the function seems to be archiving and then marking as deleted. Which is it? Can we standardize?